### DISCUSSION TITLE
Battery life significantly worse on the latest beta?

### DISCUSSION THREAD

**Discussion by jhotmann** [owntracks/android#1100] [Q&A]
My Pixel 3a (on 12 beta) appears to be losing battery much faster with 2.4.3 beta 1. No changes to other apps. Battery went from 100 to 70 over night when usually it's maybe 10% lower. Anyone else seeing similar?

**Comment by growse** [comment]
I've not tested it on 12 beta yet, but did you go straight from 2.4.2 to the current beta?

**Reply by jhotmann** [reply]
Yep. I did go ahead and revert back to 2.4.2 and my battery life went back to normal.

**Reply by growse** [reply]
How strange. Let me see if I can find a device to test out the beta on and see what's happening.

**Reply by growse** [reply]
Only thing I can see between 2.4.2 and 2.4.3-beta1 that could make a difference is https://github.com/owntracks/android/commit/758114b78fbc18597e162c171718423b1aaf3cee, which ironically aims to reduce battery usage by replacing a `Thread.sleep()` busy wait with a Java `wait()/notify()` mechanism instead. This should only have an effect if the message is failed to be delivered though, and the dispatch loop has to backoff before retrying.

Are you using the play store version, and are you connecting over MQTT or HTTP?

I'll add some debug logging and do a beta2 - would you mind trying that once it's out and we can see if that gives a clue as to what's happening?

**Reply by jhotmann** [reply]
OSS version connecting to MQTT via Websocket. I'll also run a more empirical test to compare battery usage and make sure it wasn't in my head.

**Reply by growse** [reply]
Aha, if it's the OSS version, you'll be running into a change made as a result of https://github.com/owntracks/android/issues/1096. In short, Significant monitoring mode wasn't asking for GPS location which on devices without gapps wasn't producing a location at all (no network/location service on those devices). So, there was a change to enable GPS requests on significant location updates, but obviously there's a battery impact on some devices.

Can you try raising the value of `locatorInterval` (it's in seconds) and see if that helps any?

(I should probably have a note somewhere to say that the OSS version is.... somewhat experimental and not well tested at the moment. But thanks for helping with testing it out!)

**Comment by SHA-NDY** [comment]
I also have a noticeable battery impact on my device. I have an ungoogled Device without G-Services and Gapps and without Network Location Provider. I'm using the OSS variant 2.4.5 with HTTP mode.

I'm running Android 10. Does the app require battery optimization to be turned off? Which mode is most battery friendly? I just played around a bit but couldn't figure out the best setting.

Thanks!

**Reply by growse** [reply]
The OSS variant isn't particularly well tested when it comes to battery life - the main effort into producing that flavour was around removing the Google dependencies and replacing them with aosp / oss alternatives. These may require a bit more fiddling to get battery life to be sensible.

That said, if you're seeing an actual regression (i.e. it worked great on 2.4.4 Oss but now worse on 2.4.5) then that *is* interesting. Can take a look through the diff and see what the suspects might be affecting battery usage.

**Reply by SHA-NDY** [reply]
Ah ok, unfortunately I use the app for about three days in total and can't say anything in comparison to previous versions. But I will keep track of this because the home assistant app I'm using OwnTracks with doesn't track my device at all and so I will keep using OwnTracks.

Thanks for your quick reply and have a nice weekend.

**Reply by jshadburn** [reply]
I have noticed that the OSS version keeps the gps on all the time and this appears to be what is draining the battery on my Samsung s20.  When I switched back the play store version the gps only comes on for location updates and then goes off.

**Reply by growse** [reply]
> I have noticed that the OSS version keeps the gps on all the time and this appears to be what is draining the battery on my Samsung s20.  When I switched back the play store version the gps only comes on for location updates and then goes off.

Good spot. I think there's some tweaking to be done, and likely at least 7 bugs related to how GPS is managed.

I'll try and put some time into investigating before the next release.
